pages/content.js

```diff
@@ -73,55 +45,44 @@ export default function Content({ currentCollections }) {
 
   function removeHighlight(event) {
     const span = event.target;
-    span.classList.remove(".highlight");
-    const text = span.textContent;
-    span.replaceWith(text);
-    setSelections((prevSels) =>
-      prevSels.filter(
-        (prevSel) => span.getAttribute("id") != prevSel.span.getAttribute("id")
-      )
-    );
+    const content = span.innerHTML;
+    span.replaceWith(content);
   }
 
   function highLight(event) {
-    const sel = window.getSelection();
-
-    if (!sel) return; //if no selection
-    if (sel.toString().length == 0) return;
+    const selection = window.getSelection();
+    if (!selection) return; //if no selection
+    if (selection.toString().length === 0) return;
     if (event.detail === 2) return; // double mouse click
-    const range = window.getSelection().getRangeAt(0);
-    if (
-      range.commonAncestorContainer.tagName !== undefined &&
-      range.commonAncestorContainer.tagName !== "P" &&
-      range.commonAncestorContainer.tagName !== "SPAN"
-    ) {
+
+    const range = selection.getRangeAt(0);
+    if (range.commonAncestorContainer.tagName === "DIV") {
       window.alert("Selection over several paragraph is not allowed");
       return;
     }
-    if (
-      selections.length > 0 &&
-      selections.find((prevSelection) =>
-        isSelectionOverlap(range, prevSelection.range)
-      )
-    ) {
-      window.alert("Selections should not overlap each other");
+
+    const childrenArray =
+      event.target.children === undefined ? [] : [...event.target.children];
```

Comment:
Suggestion: you could also write:
```js
const childrenArray = event.target.children || [];
```

Reply:
really? but i wanted to convert event.target.children to an array. It s a html collection

Reply:
ah sorry, then you have to spread it as well:
```js
const childrenArray = [...event.target.children] || [];
```